ui/src/logs/actions/index.ts

```diff
@@ -67,27 +162,66 @@ export const setNamespaces = (
   },
 })
 
-export const setTimeRange = (timeRange: TimeRange): SetTimeRangeAction => ({
-  type: ActionTypes.SetTimeRange,
-  payload: {
-    timeRange,
-  },
-})
+export const setTimeRange = (timeRange: TimeRange) => async dispatch => {
+  dispatch({
+    type: ActionTypes.SetTimeRange,
+    payload: {
+      timeRange,
+    },
+  })
+  dispatch(setHistogramQueryConfig())
+}
+
+export const populateNamespaces = (proxyLink: string) => async dispatch => {
+  const namespaces = await getDatabasesWithRetentionPolicies(proxyLink)
+
+  if (namespaces && namespaces.length > 0) {
+    dispatch(setNamespaces(namespaces))
+    dispatch(setNamespace(namespaces[0]))
+  }
+}
 
-export const getSourceAsync = (sourceID: string) => async dispatch => {
+export const getSourceAndPopulateNamespaces = (
+  sourceID: string
+) => async dispatch => {
   const response = await getSource(sourceID)
   const source = response.data
 
   const proxyLink = getDeep<string | null>(source, 'links.proxy', null)
 
   if (proxyLink) {
-    const namespaces = await getDatabasesWithRetentionPolicies(proxyLink)
+    dispatch(setSource(source))
+    dispatch(populateNamespaces(proxyLink))
+  }
+}
 
-    if (namespaces && namespaces.length > 0) {
-      dispatch(setNamespaces(namespaces))
-      dispatch(setNamespace(namespaces[0]))
-    }
+export const changeZoom = (timeRange: TimeRange) => async (
```

Comment:
same questions here as for all the above:
1. thunk vs action creator naming + action dispatch
1. what is the UX if no namespace or proxylink?

Reply:
then nothing changes but since we have the types of those as `Something | null` then it made since to check if we were getting to this action before anything was set.